xarray_beam/_src/core.py

```diff
@@ -255,12 +255,25 @@ def normalize_expanded_chunks(
   return result
 
 
+def _all_equal(iterator):
+  """Check if all values in a collection are equal."""
+  iterator = iter(iterator)
+  try:
+    first = next(iterator)
+  except StopIteration:
+    return True
+  return all(first == x for x in iterator)
+
+
+Chunk = Tuple[Key, Union[xarray.Dataset, Tuple[xarray.Dataset, ...]]]
```

Comment:
I would instead define this with `TypeVar` and make `DatasetToChunks` an instance of `typing.Generic`. Something like:
```python
DatasetOrDatasets = TypeVar('DatasetOrDatasets', xarray.Dataset, List[xarray.Dataset])

class DatasetToChunks(beam.PTransform, Generic[DatasetOrDatasets]):
  """Split one or more xarray.Datasets into keyed chunks."""

  def __init__(self, dataset: DatasetOrDatasets):
    ...

  def _key_to_chunks(self, key: Key) -> Iterator[DatasetOrDatasets]:
    ...
```